controllers/provider-aws/pkg/aws/client/client.go

```diff
@@ -172,3 +174,72 @@ func (c *Client) DeleteSecurityGroup(ctx context.Context, id string) error {
 	}
 	return nil
 }
+
+// DeleteObjectsWithPrefix deletes the s3 objects with the specific <prefix> from <bucket>. If it does not exist,
+// no error is returned.
+func (c *Client) DeleteObjectsWithPrefix(ctx context.Context, bucket, prefix string) error {
+	in := &s3.ListObjectsInput{
+		Bucket: aws.String(bucket),
+		Prefix: aws.String(prefix),
+	}
+	objectIDs := make([]*s3.ObjectIdentifier, 0)
+	if err := c.S3.ListObjectsPagesWithContext(ctx, in, func(page *s3.ListObjectsOutput, lastPage bool) bool {
+		for _, key := range page.Contents {
+			obj := &s3.ObjectIdentifier{
+				Key: key.Key,
+			}
+			objectIDs = append(objectIDs, obj)
+		}
+		return !lastPage
+	}); err != nil {
+		return err
+	}
+	if len(objectIDs) == 0 {
+		return nil
+	}
+
+	if _, err := c.S3.DeleteObjectsWithContext(ctx, &s3.DeleteObjectsInput{
+		Bucket: aws.String(bucket),
+		Delete: &s3.Delete{
+			Objects: objectIDs,
+		},
+	}); err != nil {
+		if aerr, ok := err.(awserr.Error); ok && aerr.Code() == s3.ErrCodeNoSuchKey {
+			return nil
+		}
+		return err
+	}
+	return nil
+}
+
+// CreateBucket creates the s3 bucket with name <bucket>. If it already exist,
+// no error is returned.
+func (c *Client) CreateBucket(ctx context.Context, bucket string) error {
```

Comment:
I'd recommend renaming this to `CreateBucketIfNotExists`

Reply:
Agreed